src/app/services/app-schema.service.ts

```diff
@@ -381,4 +381,28 @@ export class AppSchemaService {
   deleteFormListItem(event: DeleteListItemEvent): void {
     event.array.removeAt(event.index);
   }
+
+  remapAppSubmitData(data: HierarchicalObjectMap<ChartFormValue>): HierarchicalObjectMap<ChartFormValue> {
```

Comment:
What does this do?
It doesn't seem that typing is correct because both `data` and return value can be simple values like strings.

Reply:
I've rewritten this function, so it would be obvious what it does from it's signature.